### PR TITLE
Fetch info from archives.mozilla.org

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Parallel scraping
* [x] Parallel publication on kinto

### CODE REVIEW

File: jobs/README.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Listen to Pulse build and publishes records on a ``builds`` collection.
+
+Obtain Pulse user and password at https://pulseguardian.mozilla.org
+
+.. code-block:: bash
+
+    PULSEGUARDIAN_USER="my-user" PULSEGUARDIAN_PASSWORD="XXX" python2 listen_pulse.py --auth user:pass --debug
+
+
+Scrap archives
```
Comment (magopian) [inline jobs/README.rst]: I think you meant to write "scrape"

File: jobs/scrap_archives.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+               for version in versions]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_platforms(session, queue, product, version):
+    version_url = archive_url(product, version)
+    platform_folders, _ = await fetch_listing(session, version_url)
+
+    platforms = [p[:-1] for p in platform_folders]
+    platforms = [p for p in platforms if p not in ("source", "update")]
```
Comment (magopian) [inline jobs/scrap_archives.py]: I believe you put those on two separate lines to make it more readable 👍 
I would personally go one step further and comment the first line with `Strip the ending /`

File: jobs/scrap_archives.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            },
+            "systemaddons": None
+        }
+        logger.debug("Release found %s" % record["download"]["url"])
+        futures.append(queue.put(record))
+
+    return await asyncio.gather(*futures)
+
+
+async def produce(loop, queue):
```
Comment (magopian) [inline jobs/scrap_archives.py]: Add a docstring `"""Grab releases from the archives website."""`

File: jobs/scrap_archives.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    async with aiohttp.ClientSession(loop=loop) as session:
+        await fetch_products(session, queue, PRODUCTS)
+    logger.info("Scraping releases done.")
+
+
+async def consume(loop, executor, client, queue):
```
Comment (magopian) [inline jobs/scrap_archives.py]: Add a docstring `"""Store grabbed releases from the archives website in Kinto."""`

**Comment by leplatrem** [comment]
>  there's a lot of going back and forth and joining and waiting in several places.

The archives folders tree is explored in parallel, each level spawning as many promises as subfolders...

> Wasn't it possible to "merge" both the consume and produce part? 

There is a single producer with lots of parallel promises that fills a queue. There are then various consumers that consume the queue and post the records on the destination. It's one possible pattern, there might others, let's talk!

**Comment by leplatrem** [comment]
After conversing with @magopian : we use a producer/consumer pattern that makes the code more complex, but we don't take advantage of its benefits! We could at least batch requests!

Thanks dude for your clear-sightedness :)

**Pull request merged**